components/motor/gpio/motor_encoder.go

```diff
@@ -160,25 +160,14 @@ type EncodedMotor struct {
 	loop              *control.Loop
 }
 
-// EncodedMotorState is the core, non-statistical state for the motor.
-// Multiple values should be updated atomically at the same time.
-type EncodedMotorState struct {
-	regulated    bool
-	lastPowerPct float64
-	direction    float64
-}
-
 // rpmMonitor keeps track of the desired RPM and position.
-func (m *EncodedMotor) rpmMonitor(ctx context.Context, goalRPM, goalPos float64) {
-	if m.encoder == nil {
-		panic("started rpmMonitor but have no encoder")
-	}
-
+func (m *EncodedMotor) rpmMonitor(ctx context.Context, goalRPM, goalPos, direction float64) {
 	lastPos, err := m.position(ctx, nil)
 	if err != nil {
 		panic(err)
 	}
 	lastTime := time.Now().UnixNano()
+	lastPowerPct := 0.0
```

Comment:
okay so this allowed me to get rid of some other references to lastPowerPct, but it does cause the motor to briefly pause in between GoFor calls. Specifically on the RC card if you interrupt a GoFor with another GoFor, it doesn't adjust of the new speed based on the current speed it starts the whole process over from 0

Reply:
I think this is fine but can change if anyone feels strongly

Reply:
Can we grab it off of `m.real.IsPowered` so we don't have to pause? Can do in separate PR.

Reply:
no just tried it and realPowerPct always returns positive. so I need to change the other place we rely on the sign of real powerPct

Reply:
ahh I think that means I need to re-add direction to the struct

Reply:
the problem with doing an `IsPowered` call is if the motor was moving in the opposite direction initially, I think we would run into issues with rpmMonitor getting stuck moving the wrong direction again. So would prefer that change in another PR where we redo rpmMonitor entirely